### PR TITLE
fix(@aws-amplify/ui-components): Watch for formFields changes

### PR DESCRIPTION
**Pull request by jordanranz**
_Issue #, if available:_
fixes: https://github.com/aws-amplify/amplify-js/issues/5679

_Description of changes:_
- Watch for formFields update
- Handle dynamic formField value updates

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-ui-components/src/components/amplify-sign-in/amplify-sign-in.tsx [hunk 1]
```diff
@@ -192,74 +192,89 @@ export class AmplifySignIn {
...
+      },
+    });
+    this.newFormFields = [...formFieldInputs];
+  }
+
+  buildFormFields() {
+    if (this.formFields.length === 0) {
+      this.buildDefaultFormFields();
     } else {
+      const newFields = [];
```
Comment (ashika01) [inline packages/amplify-ui-components/src/components/amplify-sign-in/amplify-sign-in.tsx:258]: Nit: For consistency could we make this either `newFields` or `fields` across all the files?

File: packages/amplify-ui-components/src/components/amplify-confirm-sign-up/amplify-confirm-sign-up.tsx [hunk 2]
```diff
@@ -53,37 +53,52 @@ export class AmplifyConfirmSignUp {
...
-                {I18n.get(Translations.CONFIRM_SIGN_UP_RESEND_CODE)}
-              </amplify-button>
-            </div>
-          ),
-          handleInputChange: this.handleFormFieldInputChange('code'),
-        },
-      ];
+      this.buildDefaultFormFields();
     } else {
+      const fields = [];
```
Comment (ashika01) [inline packages/amplify-ui-components/src/components/amplify-confirm-sign-up/amplify-confirm-sign-up.tsx:95]:
```suggestion
      const newFields = [];
```

File: packages/amplify-ui-components/src/components/amplify-confirm-sign-up/amplify-confirm-sign-up.tsx [hunk 3]
```diff
@@ -53,37 +53,52 @@ export class AmplifyConfirmSignUp {
...
       this.formFields.forEach(field => {
         const newField = { ...field };
         newField['handleInputChange'] = event => this.handleFormFieldInputWithCallback(event, field);
-        this.newFormFields.push(newField);
+        fields.push(newField);
```
Comment (ashika01) [inline packages/amplify-ui-components/src/components/amplify-confirm-sign-up/amplify-confirm-sign-up.tsx:99]:
```suggestion
        newFields.push(newField);
```

File: packages/amplify-ui-components/src/components/amplify-sign-up/amplify-sign-up.tsx [hunk 4]
```diff
@@ -154,120 +154,135 @@ export class AmplifySignUp {
...
+        ];
+        break;
+    }
+  }
+
+  private buildFormFields() {
+    if (this.formFields.length === 0) {
+      this.buildDefaultFormFields();
     } else {
+      const fields = [];
```
Comment (ashika01) [inline packages/amplify-ui-components/src/components/amplify-sign-up/amplify-sign-up.tsx:266]:
```suggestion
      const newFields = [];
```

File: packages/amplify-ui-components/src/components/amplify-sign-up/amplify-sign-up.tsx [hunk 5]
```diff
@@ -154,120 +154,135 @@ export class AmplifySignUp {
...
       this.formFields.forEach(field => {
         const newField = { ...field };
         newField['handleInputChange'] = event => this.handleFormFieldInputWithCallback(event, field);
-        this.newFormFields.push(newField);
+        fields.push(newField);
```
Comment (ashika01) [inline packages/amplify-ui-components/src/components/amplify-sign-up/amplify-sign-up.tsx:270]:
```suggestion
        newFields.push(newField);
```

File: packages/amplify-ui-components/src/components/amplify-confirm-sign-up/amplify-confirm-sign-up.tsx [hunk 6]
```diff
@@ -53,37 +53,52 @@ export class AmplifyConfirmSignUp {
...
       });
+      this.newFormFields = fields;
```
Comment (ashika01) [inline packages/amplify-ui-components/src/components/amplify-confirm-sign-up/amplify-confirm-sign-up.tsx:101]:
```suggestion
      this.newFormFields = newFields;
```

File: packages/amplify-ui-components/src/components/amplify-sign-up/amplify-sign-up.tsx [hunk 7]
```diff
@@ -154,120 +154,135 @@ export class AmplifySignUp {
...
       });
+      this.newFormFields = fields;
```
Comment (ashika01) [inline packages/amplify-ui-components/src/components/amplify-sign-up/amplify-sign-up.tsx:272]:
```suggestion
      this.newFormFields = newFields;
```

File: packages/amplify-ui-components/src/components/amplify-sign-in/amplify-sign-in.tsx (hunk 1 above)
Comment (jordanranz) [reply packages/amplify-ui-components/src/components/amplify-sign-in/amplify-sign-in.tsx:258]: I will use `fields`

**Pull request merged**